pkg/catalogsourceconfig/registry.go

```diff
@@ -105,10 +105,27 @@ func (r *registry) ensureDeployment(operatorSources string) error {
 		}
 		r.log.Infof("Created Deployment %s with registry command: %s", deployment.GetName(), registryCommand)
 	} else {
-		// Update the pod specification
+		// Scale down the deployment. This is required so that we get updates
+		// from Quay during the sync cycle when packages have not been added or
+		// removed from the spec.
+		var replicas int32
+		deployment.Spec.Replicas = &replicas
+		if err = r.client.Update(context.TODO(), deployment); err != nil {
+			r.log.Errorf("Failed to scale down Deployment %s : %v", deployment.GetName(), err)
+			return err
+		}
+		r.log.Infof("Scaled down Deployment %s to 0 replicas", deployment.GetName())
+
+		// We need to get the latest version of the object after the update
+		if err := r.client.Get(context.TODO(), r.csc.key(), deployment); err != nil {
+			r.log.Errorf("Failed to get updated Deployment %s : %v", deployment.GetName(), err)
+			return err
+		}
+
+		replicas = 1
+		deployment.Spec.Replicas = &replicas
 		deployment.Spec.Template = r.newPodTemplateSpec(registryCommand)
-		err = r.client.Update(context.TODO(), deployment)
-		if err != nil {
+		if err = r.client.Update(context.TODO(), deployment); err != nil {
```

Comment:
Because we are making two updates, one happening immediately after the other. There is a chance that the controller-manager will be informed of the latest version ( which has `Replicas` set to 1, the original value ) and may not result in pod recreation. This is my understanding based on how level based system works.  I could be wrong, but just wanted to bring this up.

Reply:
I initially wrote this code with a wait for the deployment to scale down. I can go back to doing that assuming that's what you're asking for.

Reply:
Ideally we can watch pods with specific labels and do things asynchronously but we don't have time for this now, we can take a holistic approach in the future. 
For now, what you are suggesting looks to be a good stop-gap measure.

Reply:
Not sure how what we are trying to do here can be achieved by watching the pods. I would watch the deployment. Anyhow, I have https://jira.coreos.com/browse/MKTPLC-266 to track this.

Reply:
@tkashem @kevinrizza I have addressed this. Please review.